### PR TITLE
SUPPORT-1742 Send Warning for rapid RSS growth

### PR DESCRIPTION
**Pull request by ndenggg**
Added a 50MB limit for undo log

### CODE REVIEW

File: src/ee/execution/VoltDBEngine.h [hunk 1]
```diff
@@ -466,6 +466,10 @@ class __attribute__((visibility("default"))) VoltDBEngine {
                 vassert(nextUndoToken > m_currentUndoQuantum->getUndoToken());
             }
             setCurrentUndoQuantum(m_undoLog.generateUndoQuantum(nextUndoToken));
+            if (m_undoLog.getSize() > 50*1024*1024) {
+                VOLT_WARN("%s\n", "Undo log has used too much memory. Not a good thing "
+                          "from a performance perspective.");
```
Comment (zeemanhe) [inline src/ee/execution/VoltDBEngine.h]: Better to use  LogManager::getThreadLogger for proper logging.  Let me know if you need some reference of the usage.

Comment (ndenggg) [reply src/ee/execution/VoltDBEngine.h]: Thanks, I have updated this part

**Comment by zeemanhe** [comment]
did Andrew look at the message ?

File: src/ee/common/Pool.cpp [hunk 2]
```diff
@@ -75,11 +76,12 @@ void* Pool::allocate(std::size_t size) {
             return currentChunk.data();
          } else {                                       // Need to allocate a new chunk
             if (m_currentChunkIndex > m_maxChunkCount) {
-               VOLT_WARN("%s\n",
-                     "Pool had to allocate a new chunk. Not a good thing "
-                     "from a performance perspective. If you see this we need to look "
-                     "into structuring our pool sizes and allocations so the this doesn't "
-                     "happen frequently");
+                char msg[256];
```
Comment (kaikaizi) [inline src/ee/common/Pool.cpp]: `char* msg = "...";`
Or, to put it directly into where `msg` is used.

File: src/ee/execution/VoltDBEngine.h [hunk 3]
```diff
@@ -466,6 +466,12 @@ class __attribute__((visibility("default"))) VoltDBEngine {
                 vassert(nextUndoToken > m_currentUndoQuantum->getUndoToken());
             }
             setCurrentUndoQuantum(m_undoLog.generateUndoQuantum(nextUndoToken));
+            if (m_undoLog.getSize() > 50*1024*1024) {
+                char msg[128];
+                snprintf(msg, sizeof(msg), "Undo log has used too much memory. Not a good thing"
+                                           "from a performance perspective.");
+                LogManager::getThreadLogger(LOGGERID_HOST)->log(LOGLEVEL_WARN, msg);
```
Comment (kaikaizi) [inline src/ee/execution/VoltDBEngine.h]: Ditto

File: src/ee/execution/VoltDBEngine.h [hunk 4]
```diff
@@ -466,6 +466,12 @@ class __attribute__((visibility("default"))) VoltDBEngine {
                 vassert(nextUndoToken > m_currentUndoQuantum->getUndoToken());
             }
             setCurrentUndoQuantum(m_undoLog.generateUndoQuantum(nextUndoToken));
+            if (m_undoLog.getSize() > 50*1024*1024) {
```
Comment (kaikaizi) [inline src/ee/execution/VoltDBEngine.h]: Move the number constant to a static constant member of the class.

Comment (kaikaizi) [reply src/ee/execution/VoltDBEngine.h]: What is the unit of `getSize()` method's return value? Bytes? It helps to print the size into the message.

Comment (JadenW) [reply src/ee/execution/VoltDBEngine.h]: Was this warning too late? This happens after the pools already exceed the high water mark in the previous transaction.
Also maybe more helpful print out the current size.

Comment (ndenggg) [reply src/ee/execution/VoltDBEngine.h]: @JadenW Correct, it's better to give a warning at 80% and abort the txn when the total is reached

**Pull request closed without merging**